src/kernel/src/main.rs

```diff
@@ -535,3 +473,89 @@ impl Default for ExecutionEngine {
         ExecutionEngine::Llvm
     }
 }
+
+#[derive(Debug, Error)]
+enum DiscordPresenceError {
+    #[error("Failed to create Discord IPC")]
+    FailedToCreateIpc(#[source] Box<dyn Error>),
+
+    #[error("Failed to update Discord presence")]
+    FailedToUpdatePresence(#[source] Box<dyn Error>),
+}
+
+#[derive(Debug, Error)]
+enum KernelError {
+    #[error("Failed to open .kernel-debug")]
+    FailedToOpenDebug(#[source] std::io::Error),
+
+    #[error("Failed to parse .kernel-debug")]
+    FailedToParseDebug(#[from] serde_yaml::Error),
+
+    #[error("Failed to parse arguments")]
+    FailedToParseArgs(#[from] clap::Error),
+
+    #[error("Failed to open param.sfo")]
+    FailedToOpenParam(#[source] std::io::Error),
+
+    #[error("Failed to read param.sfo")]
+    FailedToReadParam(#[from] param::ReadError),
+
+    #[error("{0} has invalid title identifier.")]
```

Comment:
Same here.

Reply:
@SuchAFuriousDeath you missed this one.

Reply:
What's the issue here? That itb begines with {0}?

Reply:
[You use that yourself, twice :D](https://github.com/search?q=repo%3Aobhq%2Fobliteration%20%22%7B0%7D&type=code)

Reply:
Oh, I get it now.